eclair-core/src/main/scala/fr/acinq/eclair/router/BalanceEstimate.scala

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2021 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.router
+
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.bitcoin.{LexicographicalOrdering, Satoshi}
+import fr.acinq.eclair.MilliSatoshi.toMilliSatoshi
+import fr.acinq.eclair.router.Graph.GraphStructure.{DirectedGraph, GraphEdge}
+import fr.acinq.eclair.router.Router.{ChannelHop, PublicChannel}
+import fr.acinq.eclair.{MilliSatoshi, TimestampSecond}
+
+import scala.concurrent.duration.FiniteDuration
+
+/** Estimates the balance between a pair of nodes
+ *
+ * @param low           lower bound on the balance
+ * @param lowTimestamp  time at which the lower bound was known to be correct
+ * @param high          upper bound on the balance
+ * @param highTimestamp time at which the upper bound was known to be correct
+ * @param totalCapacity total capacity of all the channels between the pair of nodes
+ * @param halfLife      time after which the certainty of the lower/upper bounds is halved
+ */
+case class BalanceEstimate private(low: MilliSatoshi, lowTimestamp: TimestampSecond, high: MilliSatoshi, highTimestamp: TimestampSecond, totalCapacity: Satoshi, halfLife: FiniteDuration) {
```

Comment:
I think  `LiquidityEstimate` would be a much better name because it is the uncertainty in Liqudity Information that yields a problem. I know in our early work we also called it `Balance` but this term turned out to be very confusing in many conversations which is why I arrived at the point to consistantly call it `Liquidity`. 
That being said if you look at the reference implementation that I did for the stuff that I think you try to achieve here I use the terms `UncertaintyChannel` that quantifies my `belief` about the channels `Liqudity` and estimates probabilities and costs that can be used in routing decisions (in particular for the min cost flow solver) and `UncertaintyNetwork` for the entire network that maintains such information: 

you can find a very will documented notebook in with a glossary at https://github.com/renepickhardt/mpp-splitter/blob/pickhardt-payments-simulation-dev/Pickhardt-Payments-Simulation.ipynb which is currently up for review at: https://github.com/renepickhardt/mpp-splitter/pull/11